File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/customhandlers/HandlersController.java
package com.gmail.borlandlp.minigamesdtools.arena.customhandlers;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent;

import java.util.ArrayList;
import java.util.List;

public class HandlersController implements ArenaPhaseComponent {
    private List<Handler> handlers = new ArrayList<>();

    public void add(Handler handler) {
        this.handlers.add(handler);
        handler.start();
    }

    public void remove(Handler handler) {
        this.handlers.remove(handler);
        handler.stop();
    }

    @Override
    public void onInit() {

    }

    @Override
    public void beforeGameStarting() {

    }

    @Override
    public void gameEnded() {
        for (Handler handler : this.handlers) {
            try {
                handler.stop();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    public void update() {

    }

    @Override
    public void beforeRoundStarting() {

    }

    @Override
    public void onRoundEnd() {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/customhandlers/HandlersController.kt
package com.gmail.borlandlp.minigamesdtools.arena.customhandlers

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent
import java.util.*

class HandlersController : ArenaPhaseComponent {
    private val handlers: MutableList<Handler> = mutableListOf()

    fun add(handler: Handler) {
        handlers.add(handler)
        handler.start()
    }

    fun remove(handler: Handler) {
        handlers.remove(handler)
        handler.stop()
    }

    override fun onInit() {}
    override fun beforeGameStarting() {}
    override fun gameEnded() {
        for (handler in handlers) {
            try {
                handler.stop()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    override fun update() {}
    override fun beforeRoundStarting() {}
    override fun onRoundEnd() {}
}